### PR TITLE
Analytics task for sending metrics

### PR DESCRIPTION
**Pull request by ktuite**

### CODE REVIEW

File: lib/data/analytics.js [hunk 1]
```diff
@@ -124,7 +126,50 @@ const metricsTemplate = {
...
+};
+
+const convertObjectToXml = (data) => {
+  // Takes form submission data (of analytics metrics)
+  // representated as an Object and turns it into the meat
+  // of a form XML submission.
+
+  let output = '';
+  if (typeof data === 'object') {
+    for (const k in data) {
```
Comment (ktuite) [inline lib/data/analytics.js:141]: mustaches????

File: config/default.json [hunk 2]
```diff
@@ -27,7 +27,12 @@
         "clientId": "660095633112-h7bhsjenhp1agd0c4v3cmqk6bccgkdu0.apps.googleusercontent.com",
         "clientSecret": "lzu2vK1NFqqd6Y5HiN-7ByvE"
       },
-      "sentry": {}
+      "sentry": {},
+      "analytics": {
```
Comment (ktuite) [inline config/default.json:31]: fill out with the real keys

File: test/unit/data/analytics.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+describe('analytics data', () => {
+  it('should convert metrics report object to submission xml', () => {
+    const expected = "<system><num_admins><recent>11</recent><total>22</total></num_admins></system><projects><id>123</id><users><num_managers><recent>2</recent><total>2</total></num_managers></users></projects><projects><id>456</id><users><num_managers><recent>3</recent><total>3</total></num_managers></users></projects>";
+    const xml = convertObjectToXml(data);
+    xml.should.equal(expected);
+  });
+
+  it('should make a meta/uuid segment of xml', () => {
+    const xml = metaWithUuidXml().replace(/<instanceID>(.*)<\/instanceID>/, '<instanceID>1234</instanceID>');
+    xml.should.equal('<meta><instanceID>1234</instanceID></meta>');
```
Comment (ktuite) [inline test/unit/data/analytics.js:31]: xml.should.match regex?

File: test/unit/data/analytics.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    const formId = 'the-form-id';
+    const formVersion = '123';
+    const config = { email: 'test@getodk.org', organization: 'ODK' };
+    const xml = buildSubmission(formId, formVersion, data, config);
+    const expected = '<?xml version="1.0"?><data xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" id="the-form-id" version="123"><system><num_admins><recent>11</recent><total>22</total></num_admins></system><projects><id>123</id><users><num_managers><recent>2</recent><total>2</total></num_managers></users></projects><projects><id>456</id><users><num_managers><recent>3</recent><total>3</total></num_managers></users></projects><config><email>test@getodk.org</email><organization>ODK</organization></config><meta><instanceID>1234</instanceID></meta></data>';
+
+    return Submission.fromXml(xml).then((partial) => {
+      partial.xmlFormId.should.equal('the-form-id');
+      partial.instanceId.length.should.equal(41);
+      partial.xml.replace(/<instanceID>(.*)<\/instanceID>/, '<instanceID>1234</instanceID>').should.equal(expected);
```
Comment (ktuite) [inline test/unit/data/analytics.js:44]: rm

File: test/unit/data/analytics.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    });
+  });
+
+  it('should build xml when config/contact is empty', () => {
+    const formId = 'the-form-id';
+    const formVersion = '123';
+    const config = {};
+    const simpleData = { num_admins: 1 };
+    const xml = buildSubmission(formId, formVersion, simpleData, config).replace(/<instanceID>(.*)<\/instanceID>/, '<instanceID>1234</instanceID>');
+    xml.should.equal('<?xml version="1.0"?><data xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" id="the-form-id" version="123"><num_admins>1</num_admins><config></config><meta><instanceID>1234</instanceID></meta></data>');
```
Comment (ktuite) [inline test/unit/data/analytics.js:54]: check for substring with the critical part (look at odata format tests for examples)

File: test/integration/task/analytics.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      .then((res) => {
+        res.sent.should.equal(true);
+      }))));
+
+  it('should log event and full report if analytics sent successfully', testTask(({ Configs, Audits, odkAnalytics }) =>
+    Configs.set('analytics', { email: 'test@getodk.org', organization: "ODK", enabled: true })
+    .then(() => runAnalytics())
+    .then(() => Audits.getLatestByAction('analytics.submit').then((o) => o.get())
+      .then((au) => {
+        au.details.config.email.should.equal('test@getodk.org');
```
Comment (ktuite) [inline test/integration/task/analytics.js:41]: add and check success flag

File: test/integration/task/analytics.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    Configs.set('analytics', { enabled: true })
+    .then(odkAnalytics.setError({ testError: 'foo'} ))
+    .then(() => runAnalytics()
+      .then((res) => {
+        res.sent.should.equal(false);
+      }))
+    .then(() => Audits.getLatestByAction('analytics.submitFailed').then((o) => o.get())
+      .then((au) => au.details.should.eql({ testError: 'foo'})))
+    .then(() => Audits.getLatestByAction('analytics.submit')
+      .then((au) => au.isEmpty().should.equal(true))
```
Comment (ktuite) [inline test/integration/task/analytics.js:61]: change to reflect different way of logging analytics

**Review by issa-tseng — APPROVED** [review]

**Pull request merged**